src/address/envelope.rs

```diff
@@ -103,6 +103,14 @@ impl Envelope {
     pub fn from(&self) -> Option<&Address> {
         self.reverse_path.as_ref()
     }
+
+    /// Check if any of the addresses in the envelope contains non-ascii chars
+    pub(crate) fn has_non_ascii_addresses(&self) -> bool {
+        self.reverse_path
+            .iter()
+            .chain(self.forward_path.iter())
+            .any(|a| !a.is_ascii())
+    }
```

Comment:
This could be made to call `is_ascii` on both `reverse_path` and `forward_path`, since `is_ascii` on slices/strings is more optimized on some architectures.

Also it would look a bit nicer.

Reply:
Thanks for the suggestion!

Reply:
I'm not sure to understand what you mean here.

Reply:
@paolobarbolini What do you mean by calling it on both parameters?

Reply:
Ops, I guess I initially misread this. It's fine leaving it as is.